### PR TITLE
💣 DO NOT MERGE 💣 - Upload and delete image executed on update profile

### PR DESCRIPTION
**Pull request by RemcoSimonides**
- [x] Uploading and deleting image is now done when profile is updated instead of when the image is cropped

### CODE REVIEW

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/movie-tunnel.component.ts [hunk 1]
```diff
@@ -111,4 +113,54 @@ export class MovieTunnelComponent implements TunnelRoot, OnInit {
       switchMap(shouldSave => shouldSave ? this.save() : of(false))
     );
   }
+
+  private async handleUpdatePromotionalImages(moviePromotionalElementsForm: MoviePromotionalElementsForm, promotionalElements) {
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/movie-tunnel.component.ts:117]: All of this should be managed by the MediaService. You shouldn't do that here.
You should have something like : 
```typescript
const value = this.form.value;
const medias = ['promotionalElements.still_photo', 'promotionalElements.banner', ...]; // this should be exported by the movie lib so we don't have to write it several time
this.media.upload(value, medias);  // Upload your files
const updates = removePaths(value, medias); // Remove files path from update
const movie  = mergeDeep(this.query.getActive(), this.form.value); // deepmerge with current value
await this.service.update(movie); // Update everything except media fields.
```

File: apps/festival/festival/src/app/dashboard/tunnel/media-image/media-image.component.html [hunk 2]
```diff
@@ -48,7 +48,7 @@ <h3 type>Poster</h3>
         <ng-container *ngFor="let control of stillPhoto.controls; let i = index; let last = last">
           <span class="still mat-body-2">Still # {{i + 1}}</span>
           <drop-cropper
-            [storagePath]="stillPath"
+            [storagePath]="'movies/' + movie.id + '/promotionalElements.still_photo.' + i + '.media'"
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/dashboard/tunnel/media-image/media-image.component.html:51]: Same as above

File: apps/festival/festival/src/app/dashboard/tunnel/movie-tunnel.component.ts [hunk 3]
```diff
@@ -100,4 +102,54 @@ export class MovieTunnelComponent implements TunnelRoot, OnInit {
       switchMap(shouldSave => shouldSave ? this.save() : of(false))
     );
   }
+
+  private async handleUpdatePromotionalImages(moviePromotionalElementsForm: MoviePromotionalElementsForm, promotionalElements) {
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/dashboard/tunnel/movie-tunnel.component.ts:106]: Same as above

File: libs/ui/src/lib/media/cropper/cropper/cropper.component.html [hunk 4]
```diff
@@ -42,7 +42,7 @@
   <article class="show" *ngSwitchCase="'show'" @finalZoom>
     <ng-container>
       <div class="show">
-        <img class="croppedImage" [src]="previewUrl | async" />
+        <img class="croppedImage" [src]="croppedImage || (previewUrl | async)" />
```
[the remark this answers is not in the document]
Comment (RemcoSimonides) [reply libs/ui/src/lib/media/cropper/cropper/cropper.component.html:45]: There is no previewUrl until it has been uploaded to storage.

**Pull request closed without merging**